onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -768,45 +770,244 @@
+    std::ofstream& outfile) {
   QnnProfile_EventData_t event_data;
   auto result = qnn_interface_.profileGetEventData(profile_event_id, &event_data);
-  ORT_RETURN_IF(QNN_PROFILE_NO_ERROR != result, "Failed to get provile event data.");
+  QnnProfile_Error_t errorCode = static_cast<QnnProfile_Error_t>(result & 0xFFFF);
+  ORT_RETURN_IF(QNN_PROFILE_NO_ERROR != result, "Failed to get profile event data: " + std::string(QnnProfileErrorToString(errorCode)));
+
+  std::string message = GetEventTypeString(event_data.type);
+  std::string unit = GetUnitString(event_data.unit);
+
+  outfile << "UNKNOWN"
+          << ","
+          << message << ","
+          << event_data.value << ","
+          << unit << ","
+          << "BACKEND"
+          << ","
+          << eventLevel << ","
+          << (event_data.identifier ? event_data.identifier : "NULL") << "\n";
 
-  LOGS(*logger_, VERBOSE) << "Profiling Event Info - Event Type: " << event_data.type
-                          << ", Event Value: " << event_data.value
-                          << ", Event Identifier: " << event_data.identifier
-                          << ", Event Unit: " << event_data.unit;
+  return Status::OK();
+}
+
+Status QnnBackendManager::ExtractProfilingEventExtended(
+    QnnProfile_EventId_t profile_event_id,
+    std::string eventLevel,
+    std::ofstream& outfile) {
+  QnnProfile_ExtendedEventData_t event_data_extended;
+  auto resultGetExtendedEventData = qnn_interface_.profileGetExtendedEventData(profile_event_id, &event_data_extended);
+  QnnProfile_Error_t errorCode = static_cast<QnnProfile_Error_t>(resultGetExtendedEventData & 0xFFFF);
+  ORT_RETURN_IF(QNN_PROFILE_NO_ERROR != errorCode, "Failed to get profile event data: " + std::string(QnnProfileErrorToString(errorCode)));
+
+  std::string message = GetEventTypeString(event_data_extended.v1.type);
+  std::string unit = GetUnitString(event_data_extended.v1.unit);
+
+  if (event_data_extended.version == QNN_PROFILE_DATA_VERSION_1) {
+    outfile << event_data_extended.v1.timestamp << ","
+            << message << ","
+            << ExtractQnnScalarValue(event_data_extended.v1.value) << ","
+            << unit << ","
+            << "BACKEND"
+            << ","
+            << eventLevel << ","
+            << (event_data_extended.v1.identifier ? event_data_extended.v1.identifier : "NULL") << "\n";
+  }
 
   return Status::OK();
 }
 
+std::string QnnBackendManager::GetUnitString(QnnProfile_EventUnit_t unitType) {
+  // Interpret the unit of measurement using the macros
+  switch (unitType) {
+    case QNN_PROFILE_EVENTUNIT_MICROSEC:
+      return "US";
+    case QNN_PROFILE_EVENTUNIT_BYTES:
+      return "BYTES";
+    case QNN_PROFILE_EVENTUNIT_CYCLES:
+      return "CYCLES";
+    case QNN_PROFILE_EVENTUNIT_COUNT:
+      return "COUNT";
+    case QNN_PROFILE_EVENTUNIT_OBJECT:
+      return "OBJECT";
+    case QNN_PROFILE_EVENTUNIT_BACKEND:
+      return "BACKEND";
+    default:
+      return "UNKNOWN";
+  }
+}
```

Comment:
create a static loop up table using unordered_map, so that we can return a const std::sting&

Reply:
Fixed although leaving them as private member funcs as could not find better place for them right now